packages/nodes-base/credentials/WhatsAppOAuth2Api.credentials.ts

```diff
@@ -0,0 +1,70 @@
+import type { ICredentialTestRequest, ICredentialType, INodeProperties } from 'n8n-workflow';
+
+export class WhatsAppOAuth2Api implements ICredentialType {
+	name = 'whatsAppOAuth2Api';
+
+	extends = ['oAuth2Api'];
+
+	displayName = 'WhatsApp OAuth2 API';
+
+	documentationUrl = 'whatsApp';
+
+	properties: INodeProperties[] = [
+		{
+			displayName: 'Grant Type',
+			name: 'grantType',
+			type: 'hidden',
+			default: 'clientCredentials',
+		},
+		{
+			displayName: 'Authorization URL',
+			name: 'authUrl',
+			type: 'hidden',
+			default: 'https://www.facebook.com/v19.0/dialog/oauth',
+			required: true,
+		},
+		{
+			displayName: 'Access Token URL',
+			name: 'accessTokenUrl',
+			type: 'hidden',
+			default: 'https://graph.facebook.com/v19.0/oauth/access_token',
+			required: true,
+		},
+		{
+			displayName: 'Scope',
+			name: 'scope',
+			type: 'hidden',
+			default: 'whatsapp_business_management whatsapp_business_messaging',
+		},
+		{
+			displayName: 'Auth URI Query Parameters',
+			name: 'authQueryParameters',
+			type: 'hidden',
+			default: '',
+		},
+		{
+			displayName: 'Authentication',
+			name: 'authentication',
+			type: 'hidden',
+			default: 'header',
+		},
+	];
+
+	test: ICredentialTestRequest = {
+		request: {
+			baseURL: 'https://graph.facebook.com/v19.0',
+			url: '/',
+			ignoreHttpStatusErrors: true,
+		},
+		rules: [
+			{
+				type: 'responseSuccessBody',
+				properties: {
+					key: 'error.type',
+					value: 'OAuthException',
+					message: 'Invalid access token',
+				},
+			},
+		],
+	};
```

Comment:
why http error is concealed here? reason we have this test to validate user credentials information

Reply:
This is a copy of one of the other credentials, didn't change it. So I do not know. I do not know what this is doing.